common/src/main/java/org/apache/uniffle/common/util/ByteBufUtils.java

```diff
@@ -80,10 +80,10 @@ public static final ByteBuf readSlice(ByteBuf from) {
     return from.retain().readSlice(length);
   }
 
-  public static final byte[] readBytes(ByteBuf buf) {
+  public static final byte[] readBytes(ByteBuf byteBuf) {
+    ByteBuf buf = byteBuf.duplicate();
```

Comment:
The duplicate just is a reference count that is low cost, anyway this is a good improvement.